### PR TITLE
feat(gateway): Add default HTTP headers to endpoints

### PR DESCRIPTION
**Pull request by brasseld**
Closes gravitee-io/issues#1740

### CODE REVIEW

File: gravitee-gateway-http/src/main/java/io/gravitee/gateway/http/endpoint/HttpEndpointFactory.java [hunk 1]
```diff
@@ -56,6 +55,10 @@ public boolean support(EndpointType endpointType) {
             endpoint.getHttpProxy().setPassword(convert(endpoint.getHttpProxy().getPassword()));
         }
 
+        // Default HTTP headers
+        if (endpoint.getHeaders() != null && endpoint.getHeaders().isEmpty()) {
```
[the remark this answers is not in the document]
Comment (NicolasGeraud) [reply gravitee-gateway-http/src/main/java/io/gravitee/gateway/http/endpoint/HttpEndpointFactory.java]: or perhaps you just need to test if not null and then replaceall
